tests/python/pants_test/engine/test_engine.py

```diff
@@ -102,6 +104,7 @@ async def rule_one_function(i: Input) -> Beta:
     a = Alpha()
     o = await Get[Omega](Alpha, a)
     b = await Get[Beta](Omega, o)
+    time.sleep(3)
```

Comment:
I see how `sleep()` is used, but be warned that this is one of the biggest causes of flaky tests. Please be sure to run this test multiple times (e.g. 20 times) to ensure this does not flake. You'll need to use the V1 test runner with `--test-pytest-cache-ignore`, because the V2 test runner always uses the cache.